### PR TITLE
Fix all clazy warnings and enable it in the CI

### PR DESCRIPTION
**Pull request by FlexW**
Related issue: https://github.com/nextcloud/desktop/issues/3412

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -92,6 +92,40 @@ trigger:
...
+    - cd /drone/build
+    - cmake -GNinja -DCMAKE_C_COMPILER=clang-10 -DCMAKE_CXX_COMPILER=clazy -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DECM_ENABLE_SANITIZERS=address -DCMAKE_CXX_FLAGS=-Werror ../src
+- name: compile
+  image: ghcr.io/nextcloud/continuous-integration-client:client-5.15-2
+  volumes:
+    - name: build
+      path: /drone/build
+  commands:
+    - cd /drone/build
+    - CLAZY_IGNORE_DIRS=".*3rdparty.*|.*build.*" ninja
```
Comment (mgallien) [inline .drone.yml:115]: better use this [clazy-standalone](https://github.com/KDE/clazy#clazy-standalone-and-json-database-support)
that should be faster than doing a real compilation (that should not be needed we already compile with clang before

File: .drone.yml [hunk 2]
```diff
@@ -92,6 +92,40 @@ trigger:
...
+
+steps:
+- name: cmake
+  image: ghcr.io/nextcloud/continuous-integration-client:client-5.15-2
+  volumes:
+    - name: build
+      path: /drone/build
+  commands:
+    - cd /drone/build
+    - cmake -GNinja -DCMAKE_C_COMPILER=clang-10 -DCMAKE_CXX_COMPILER=clazy -DCMAKE_BUILD_TYPE=Debug -DBUILD_UPDATER=ON -DBUILD_TESTING=1 -DECM_ENABLE_SANITIZERS=address -DCMAKE_CXX_FLAGS=-Werror ../src
```
Comment (mgallien) [inline .drone.yml:107]: you should enable the generation of compilation database

File: src/common/syncjournaldb.h [hunk 3]
```diff
@@ -282,8 +282,9 @@ class OCSYNC_EXPORT SyncJournalDb : public QObject
      */
     struct OCSYNC_EXPORT PinStateInterface
     {
+        PinStateInterface(SyncJournalDb *db);
         PinStateInterface(const PinStateInterface &) = delete;
-        PinStateInterface(PinStateInterface &&) = delete;
+        PinStateInterface(PinStateInterface &&other);
```
Comment (mgallien) [inline src/common/syncjournaldb.h:287]: why those changes ?

File: src/common/syncjournaldb.cpp [hunk 4]
```diff
@@ -2281,6 +2281,17 @@ void SyncJournalDb::markVirtualFileForDownloadRecursively(const QByteArray &path
...
 
+SyncJournalDb::PinStateInterface::PinStateInterface(SyncJournalDb *db)
+    : _db(db)
+{
+}
+
+SyncJournalDb::PinStateInterface::PinStateInterface(PinStateInterface &&other)
+    : _db(other._db)
+{
+    other._db = nullptr;
```
Comment (mgallien) [inline src/common/syncjournaldb.cpp:2292]: use `std::swap`

File: src/gui/accountmanager.cpp [hunk 5]
```diff
@@ -80,7 +80,8 @@ bool AccountManager::restore()
         return true;
     }
 
-    for (const auto &accountId : settings->childGroups()) {
+    const auto &childGroups = settings->childGroups();
+    for (const auto &accountId : childGroups) {
```
Comment (mgallien) [inline src/gui/accountmanager.cpp:84]: `for (const auto &accountId : qAsConst(settings->childGroups())) {`

**Review by mgallien — CHANGES_REQUESTED** [review]
a few comments
please for example for `qAsConst` apply it everywhere

File: .drone.yml (hunk 2 above)
Comment (FlexW) [reply .drone.yml:107]: see my comment above.

File: src/common/syncjournaldb.h (hunk 3 above)
Comment (FlexW) [reply src/common/syncjournaldb.h:287]: Otherwise, it is impossible to use `setForPath` properly.

**Pull request closed without merging**